umbraco/Umbraco-CMS
# Fix preview issue for new (empty) variants

### Prerequisites


If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/12884

### Description

When attempting to preview a newly created language variant which has validation errors that prevents it from being saved, the validation kicks in _after_ the initial launch of the preview window. This causes a few problems:

1. The preview window remains blank and is now broken for any future preview attempts from other language variants (as described in the linked issue)
2. The editor likely won't see the validation errors, as the blank preview window blocks the backoffice.

The fix is two-fold:

1. Make sure the preview window is only launched once validation has completed (for save events).
2. Ensure that "Save and preview" _only_ attempts to save the active variant, to avoid validation interference from other variants.  
    - As-is the "Save and preview" action will save all edited variants - without prompting the editor to do so (!), which is actually a rather unwanted behaviour. In comparison, the "Save..." action explicitly prompts the editor to pick which of edited variants to save. 

### Testing this PR

1. Create a new language variant piece of content and save only the first variant. 
2. Hit "Save and preview" and verify that preview works.
3. Switch to another language variant and hit "Save and preview" immediately.
4. Verify that validation kicks in, prompting you (subtly) to fill in the variant name (and any mandatory props).
5. Switch back to the first variant and verify that preview still works.
6. Switch back to the other variant again, fill out mandatory data and verify that "Save and preview" works.

![save-preview-12884](https://user-images.githubusercontent.com/7405322/203261416-4a8a2777-83c0-4f04-90ef-f7cfec28f140.gif)
